rust/src/combinators/measure_cast/zCDP_to_approxDP/cdp_delta/mod.rs

```diff
@@ -0,0 +1,101 @@
+use std::ops::Neg;
+
+use num::Zero;
+
+use crate::{
+    error::Fallible,
+    traits::{InfAdd, InfDiv, InfExp, InfLn1P, InfMul, InfSub},
+};
+
+#[cfg(test)]
+mod test;
+
+pub(crate) fn cdp_delta(rho: f64, eps: f64) -> Fallible<f64> {
+    if rho.is_sign_negative() {
+        return fallible!(FailedMap, "rho must be non-negative");
+    }
+
+    if eps.is_sign_negative() {
+        return fallible!(FailedMap, "epsilon must be non-negative");
+    }
+
+    if rho.is_zero() || eps.is_infinite() {
+        return Ok(0.0);
+    }
+
+    if rho.is_infinite() {
+        return Ok(1.0);
+    }
```

Comment:
Is the special casing for infinity something that should also show up on the Python side? Should there be python tests with `float('inf')`?

Reply:
It is tested, maybe surprisingly-- it's the case where scale is zero. Zero noise means the distributions are disjoint, resulting in infinite rho.